drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1118 @@
+		regdata |= TMAG5273_TRIGGER_MODE_INT;
+	}
+
+	if (drv_cfg->operation_mode == TMAG5273_DT_OPER_MODE_CONTINUOUS) {
+		regdata |= TMAG5273_OPERATING_MODE_CONTINUOUS;
+	}
+
+	/* Note: I2C glitch filter enabled by default */
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_DEVICE_CONFIG_2, regdata);
+	if (retval < 0) {
+		LOG_ERR("error setting DEVICE_CONFIG_2");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief sets the \c SENSOR_CONFIG_1 and \c SENSOR_CONFIG_2 registers
+ *
+ * @param drv_cfg configuration of the TMAG5273 instance
+ *
+ * @retval 0 if everything was okay
+ * @retval -EIO on communication errors
+ */
+static inline int tmag5273_init_sensor_settings(const struct tmag5273_config *drv_cfg)
+{
+	int retval;
+	uint8_t regdata;
+
+	/* REG_SENSOR_CONFIG_1 */
+	regdata = 0;
+
+	if (drv_cfg->x_axis) {
+		regdata |= TMAG5273_MAG_CH_EN_X;
+	}
+	if (drv_cfg->y_axis) {
+		regdata |= TMAG5273_MAG_CH_EN_Y;
+	}
+	if (drv_cfg->z_axis) {
+		regdata |= TMAG5273_MAG_CH_EN_Z;
+	}
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_1, regdata);
+	if (retval < 0) {
+		LOG_ERR("error setting SENSOR_CONFIG_1");
+		return -EIO;
+	}
+
+	/* REG_SENSOR_CONFIG_2 */
+	regdata = 0;
+
+	if (drv_cfg->ch_mag_gain_correction == TMAG5273_DT_CORRECTION_CH_2) {
+		regdata |= TMAG5273_MAG_GAIN_CORRECTION_CH_2;
+	}
+
+	switch (drv_cfg->angle_magnitude_axis) {
+	case TMAG5273_DT_ANGLE_MAG_XY:
+		regdata |= TMAG5273_ANGLE_EN_XY;
+		break;
+	case TMAG5273_DT_ANGLE_MAG_YZ:
+		regdata |= TMAG5273_ANGLE_EN_YZ;
+		break;
+	case TMAG5273_DT_ANGLE_MAG_XZ:
+		regdata |= TMAG5273_ANGLE_EN_XZ;
+		break;
+	case TMAG5273_DT_ANGLE_MAG_RUNTIME:
+	case TMAG5273_DT_ANGLE_MAG_NONE:
+		__fallthrough;
+	default:
+		regdata |= TMAG5273_ANGLE_EN_POS;
+		break;
+	}
+
+	if (drv_cfg->meas_range == TMAG5273_DT_AXIS_RANGE_LOW) {
+		regdata |= TMAG5273_XYZ_MEAS_RANGE_LOW;
+	} else {
+		regdata |= TMAG5273_XYZ_MEAS_RANGE_HIGH;
+	}
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2, regdata);
+	if (retval < 0) {
+		LOG_ERR("error setting SENSOR_CONFIG_2");
+		return -EIO;
+	}
+
+	/* REG_T_CONFIG */
+	regdata = 0;
+
+	if (drv_cfg->temperature) {
+		regdata |= TMAG5273_T_CH_EN_ENABLED;
+	}
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_T_CONFIG, regdata);
+	if (retval < 0) {
+		LOG_ERR("error setting SENSOR_CONFIG_2");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief initialize a TMAG5273 sensor
+ *
+ * @param dev handle to the device
+ *
+ * @retval 0 on success
+ * @retval -EINVAL if bus label is invalid
+ * @retval -EIO on communication errors
+ */
+static int tmag5273_init(const struct device *dev)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+	int retval;
+	uint8_t regdata;
+
+	if (!i2c_is_ready_dt(&drv_cfg->i2c)) {
+		LOG_ERR("could not get pointer to TMAG5273 I2C device");
+		return -ENODEV;
+	}
+
+	if (drv_cfg->trigger_conv_via_int) {
```

Comment:
would it make sense to check for `int_gpio.port != NULL` instead? makes more sense since the block relies on that value to not be null, I guess one could set the gpio and not enable interrupt conversion and then it would get initialized and not used, but that's probably okay, better than letting it float anyway

Reply:
The idea is to not rely on the definition of `int-gpios` to figure out what kind of conversion trigger is used.

This is useful later on when interrupt handling is added, so that the definition in the binding regarding the conversion trigger setup does not change.